library/x509.c

```diff
@@ -354,6 +354,8 @@ static int x509_get_attr_type_value( unsigned char **p,
             MBEDTLS_ASN1_CONSTRUCTED | MBEDTLS_ASN1_SEQUENCE ) ) != 0 )
         return( MBEDTLS_ERR_X509_INVALID_NAME + ret );
 
+    end = *p + len;
```

Comment:
Does this make the below assignment of `oid->tag = **p;` wrong? since before we had `oid->tag` pointing to the beginning of the tag. Now, we have it pointing to the end of the tag.

Consider setting `oid->tag` to `MBEDTLS_ASN1_CONSTRUCTED | MBEDTLS_ASN1_SEQUENCE` explicitly.

Reply:
I don't follow - the line modifies `end`, but not `*p`, hence not `oid->tag`. Am I missing something?

Reply:
`oid->tag` will necessarily be `MBEDTLS_ASN1_OID` in the end, and one could move `oid->tag = **p` to `oid->tag = MBEDTLS_ASN1_OID` once the `mbedtls_asn1_get_tag()` call succeeded. Maybe the latter saves a few bytes, so let's do it.

Reply:
Would you be OK with leaving this change to a separate X.509 PR (there are enough waiting), so that we don't undo @k-stachowiak's review in case there's nothing else to be changed?

Reply:
Yup, I'm misreading the code. `p` hasn't moved.